apps/bfd-pipeline/bfd-pipeline-shared-utils/src/main/java/gov/cms/bfd/pipeline/sharedutils/PipelineApplicationState.java

```diff
@@ -0,0 +1,161 @@
+    this.metrics = metrics;
+    this.pooledDataSource = pooledDataSource;
+    this.entityManagerFactory = entityManagerFactory;
+  }
+
+  /**
+   * This is marked <code>public</code> only for use in tests; non-test code that needs a {@link
+   * DataSource} should get it from the {@link PipelineApplicationState}.
+   *
+   * @param dbOptions the {@link DatabaseOptions} to use
+   * @param metrics the {@link MetricRegistry} to use
+   * @return a {@link HikariDataSource} for the BFD database
+   */
+  private static HikariDataSource createPooledDataSource(
+      DatabaseOptions dbOptions, MetricRegistry metrics) {
+    HikariDataSource pooledDataSource = new HikariDataSource();
+
+    pooledDataSource.setJdbcUrl(dbOptions.getDatabaseUrl());
+    pooledDataSource.setUsername(dbOptions.getDatabaseUsername());
+    pooledDataSource.setPassword(dbOptions.getDatabasePassword());
+    pooledDataSource.setMaximumPoolSize(dbOptions.getMaxPoolSize());
+    pooledDataSource.setRegisterMbeans(true);
+    pooledDataSource.setMetricRegistry(metrics);
+
+    return pooledDataSource;
+  }
+
+  /**
+   * This is marked <code>public</code> only for use in tests; non-test code that needs a {@link
+   * DataSource} should get it from the {@link PipelineApplicationState}.
+   *
+   * @param dataSource a non-pooled {@link DataSource} for the application's DB (which this will use
+   *     to create {@link #getPooledDataSource()})
+   * @param maxPoolSize the {@link DatabaseOptions#getMaxPoolSize()} value to use
+   * @param metrics the {@link MetricRegistry} to use
+   * @return a {@link HikariDataSource} for the BFD database
+   */
+  private static HikariDataSource createPooledDataSource(
+      DataSource dataSource, int maxPoolSize, MetricRegistry metrics) {
+    HikariDataSource pooledDataSource = new HikariDataSource();
+
+    pooledDataSource.setDataSource(dataSource);
+    pooledDataSource.setMaximumPoolSize(maxPoolSize);
+    pooledDataSource.setRegisterMbeans(true);
+    pooledDataSource.setMetricRegistry(metrics);
+
+    return pooledDataSource;
+  }
+
+  /**
+   * This is marked <code>public</code> only for use in tests; non-test code that needs a {@link
+   * EntityManagerFactory} should get it from the {@link PipelineApplicationState}.
+   *
+   * @param pooledDataSource the JDBC {@link DataSource} for the application's database
+   * @return a JPA {@link EntityManagerFactory} for the application's database
+   */
+  private static EntityManagerFactory createEntityManagerFactory(DataSource dataSource) {
+    /*
+     * The number of JDBC statements that will be queued/batched within a single transaction. Most
+     * recommendations suggest this should be 5-30. Paradoxically, setting it higher seems to
+     * actually slow things down. Presumably, it's delaying work that could be done earlier in a
+     * batch, and that starts to cost more than the extra network roundtrips.
+     */
+    int jdbcBatchSize = 10;
```

Comment:
Looks like this was a static final class member earlier which seems preferable. Any particular reason for the change? I think it at least should be final.

Reply:
I try not to make things constants unless they're actually referenced by more than one block, as the indirection hurts readability. But sure, we can mark it `final` -- why not. Done.

Reply:
Ditto.